### PR TITLE
136 bank headword list

### LINKED ISSUE #136: Bank — headword list

### PR DESCRIPTION
**Pull request by hangler**
Fix #136

### CODE REVIEW

File: app/routes/bank/edit/$citationId.tsx [hunk 1]
```diff
@@ -75,7 +74,7 @@ export const action = async ({ request, params }: ActionArgs) => {
       : 0,
   }
 
-  const updatedCitation = await updateCitation(citationFields)
+  await updateCitation(citationFields)
```
Comment (Shawk031) [inline app/routes/bank/edit/$citationId.tsx:77]: Not sure if it's here, but when you try to update a citation for which there is no headword / the headword doesn't exist, the error isn't caught

Comment (Shawk031) [reply app/routes/bank/edit/$citationId.tsx:77]: like for the entries for "b" for example

File: app/routes/bank/headword-list/$initialLetter.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+  const headwords = await prisma.$queryRaw<
+    {
+      id: number
+      headword: string
+      count: number
+    }[]
+  >`SELECT headword, count('c.id') AS count FROM citation c INNER JOIN headword h ON c.headword_id = h.id INNER JOIN user u ON c.user_id = u.id WHERE UPPER(headword) LIKE UPPER(${query}) GROUP BY UPPER(headword), h.id ORDER BY UPPER(headword)`
+
+  console.log("headwords", headwords)
```
Comment (Shawk031) [inline app/routes/bank/headword-list/$initialLetter.tsx:19]: also if you're gonna make any changes would probably be nice to remove these logs. Though if you're not making any changes I'll be cleaning these up sometime soon hopefully

**Review by Shawk031 — APPROVED** [review]
some minor adjustments

File: app/routes/bank/headword-list/$initialLetter.tsx (hunk 2 above)
Comment (hangler) [reply app/routes/bank/headword-list/$initialLetter.tsx:19]: Good catch. Thanks.

File: app/routes/bank/edit/$citationId.tsx (hunk 1 above)
Comment (hangler) [reply app/routes/bank/edit/$citationId.tsx:77]: Yeah there's a whole lotta exception handling that needs doing 😅

Comment (hangler) [reply app/routes/bank/edit/$citationId.tsx:77]: That said, I can't seem to reproduce the error. The new headword should be created by the call to `findOrCreateHeadword`. I'm going to close this conversation, but file a new issue for this.

File: app/routes/bank/citations/$headword.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+export function CatchBoundary() {
+  const caught = useCatch()
+
+  if (caught.status === 404) {
+    return <div>Entry not found</div>
+  }
+
+  throw new Error(`Unexpected caught response with status: ${caught.status}`)
+}
```
[the remark this answers is not in the document]
Comment (hangler) [reply app/routes/bank/citations/$headword.tsx:56]: Oh, one issue — I forgot to mention that I updated the version of Remix in package.json, though I forget if it's on this branch. Can you try doing an `npm install` to refresh the version numbers?

**Pull request merged**